### DISCUSSION TITLE
Powertoys v2

### DISCUSSION THREAD

**Discussion by lin-ycv** [lin-ycv/EverythingPowerToys#153] [Announcements]
Microsoft have announced they're working on Powertoys v2:
https://x.com/cinnamon_msft/status/1871966829935304872
And yes, EPT will be ported over to v2, their team has established contact with me regarding this.

**Comment by hlaueriksson** [comment]
What did you learn from the team?

**Reply by lin-ycv** [reply]
We've just established contact, so not much atm. 
Most of their them is on holiday, so actual hands on and testing from me will have to wait until after the holiday season. 
What I do know is that v2 has been in the works for a while now, and during that time they've made and tested a EPT port. 
I'll share more once I actually get hands on time and have used it thoroughly.

**Reply by lin-ycv** [reply]
So I've done some preliminary tests on v2 of powertoys run, it's a complete rebuilt and quite a lot more complex than the current implementation (v2 also offers a lot more features/flexibility, tho some currently available feature are missing/WIP).
I don't think I'm able to create a fully functioning port in the short term, but I also think MS won't be shipping it that soon, there's still some "bugs".

**Reply by hlaueriksson** [reply]
Did you get access to the source code of v2?

**Reply by lin-ycv** [reply]
Yes, just like v1, the whole thing is OS. They haven't publicly stated where/how to test it, because they're still ironing it out.

**Comment by lin-ycv** [comment]
**Update** I think Command Palette (CmdPal / Run v2) will be released _soon_
I have a "working" port that uses Everything's SDK2
![image](https://github.com/user-attachments/assets/d4d8ffcd-2d42-4365-9bfb-d36245593dd2)
There are still "bugs", and from what I understand Run and Command Palette will co-exist for a while; so there'll be potentially 5 different EPT variation for next release:

1. x64 (Run)
2. ARMx64 (Run)
3. x64 SDK3 (Run)
4. x64 (CmdPal)
5. x64 SDK3 (CmdPal)

Those that are interested in the new variant, source is here: https://github.com/lin-ycv/EverythingCommandPalette

**Reply by freeshrugsxd** [reply]
Is it possible to get results from Everything in the start window of cmd pal? Or do we have to always select Everything first? I would like to open cmd pal and just start typing to get my search results

**Reply by lin-ycv** [reply]
@freeshrugsxd moving this to https://github.com/lin-ycv/EverythingCommandPalette/issues/1
